packages/server-wallet/src/models/ledger-request.ts

```diff
@@ -1,16 +1,25 @@
+import {BN} from '@statechannels/wallet-core';
 import {Model, TransactionOrKnex} from 'objection';
 
-import {Bytes32} from '../type-aliases';
+import {Uint256, Destination, Bytes32} from '../type-aliases';
 
 export type LedgerRequestStatus =
+  | 'queued'
+  | 'cancelled'
   | 'pending' // Request added to DB to be handled by ProcessLedgerQueue
```

Comment:
Maybe this is the reason we need both queued and pending statuses? https://github.com/statechannels/statechannels/blob/tc/new-ledger-algorithm/packages/server-wallet/src/protocols/__test__/ledger-manager.test.ts#L173

Reply:
I think it isn't actually important for the algorithm, as it's deducible from the ledger states, and I'm pretty sure I always do this.

It might be nice to be able to communicate this information to the objective and/or user though. It's effectively "we're just waiting for your opponent to sign now". Initially I thought that objectives might behave differently according to whether the request was pending, as once a request is pending it's out of the wallets control whether it gets accepted or not, so the request can't be withdrawn. However, that can be fragile (a pending request might revert to queued; a race condition might mean that a queued request becomes pending), and the just-send-a-cancelling-defund approach handles these issues optimally and robustly.

Overall, I think I'd be inclined to leave it at this point as it (a) works, (b) isn't doing any harm and (c) potentially provides a bit of extra debugging/UX info, even though it isn't strictly necessary.